src/service/performance-impl.js

```diff
@@ -233,6 +269,48 @@ export class Performance {
     observer.observe({entryTypes: entryTypesToObserve});
   }
 
+  /**
+   * Bind to various page lifecycle events to capture visibility state changes.
+   * @see https://developers.google.com/web/updates/2018/07/page-lifecycle-api
+   */
+  registerLayoutJankHandler_() {
+    // The visibility state of the document.
+    let visibilityState = getVisibilityState(this.win.document);
+    // Whether the `lj` metric was sent.
+    let sentFirstJankScore = false;
+    // Whether the `lj-2` metric was sent.
+    let sentSecondJankScore = false;
+
+    const handleVisibilityStateChange = nextState => {
+      const previousState = visibilityState;
+      if (previousState === nextState) {
+        return;
+      }
+      visibilityState = nextState;
+
+      if (nextState !== 'hidden') {
+        return;
+      }
+
+      if (!sentFirstJankScore) {
+        this.tickDelta('lj', this.aggregateJankScore_);
+        this.flush();
+        sentFirstJankScore = true;
+        return;
+      }
+      if (!sentSecondJankScore) {
+        this.tickDelta('lj-2', this.aggregateJankScore_);
+        this.flush();
+        sentSecondJankScore = true;
+        return;
+      }
+    };
+
+    ['pageshow', 'focus', 'blur', 'visibilitychange', 'resume'].forEach(eventName => {
```

Comment:
You're listening for `focus` and `blur` but then you're not doing anything with the `active` and `passive` states. If you don't care about those states then you probably can omit the `focus` and `blur` events from this list.

Also, if you need to detect a change to the `hidden` state in Safari, then you'll need to also add the `beforeunload` event (but only in Safari). See the comment here for an explanation: https://github.com/GoogleChromeLabs/page-lifecycle/blob/41ed9db/src/Lifecycle.mjs#L156-L171

Reply:
Done.

If we only care about the change of a page from a hidden state ... it seems like we could just listen to the `visibilitychange` event here, and avoid `pageshow` and `resume` as well?

Reply:
I assumed you were using pageshow to detect coming out of bfcache. You'd have to test whether those browsers fire visibilitychange in that situation (I suspect they don't, so you may want to keep listening to it).

For `resume`, you'd also have to check to see whether Chrome fires `visibilitychange` after `resume`. It's should be, so if it's not could you file a bug?

Reply:
I might not have explained the use case clearly. We're interested in recording layout jank just when the user leaves the page. So checking the `document.visibilityState` whenever the `visibilitychange` event is fired seems sufficient for Chrome in my testing, and adding the `beforeunload` event for Safari as you mentioned. 

Although a `resume` event may be fired when the document is in the hidden state, it seems this would be the result of a page transitioning from frozen to hidden, which I believe would happen when a user re-enters a backgrounded tab?

Reply:
I see. I think I was originally assuming you were removing the event listener on `hidden` and then re-adding it for the 2nd jank score, but looking at it again, that's clearly not what you're doing. In that case, yes, just `visibilitychange` (and `beforeunload` for Safari) should be sufficient.